### PR TITLE
Add random slice mutation

### LINKED ISSUE #392: Implement Random Slice Mutations
### Description
- implement mutation which randomly takes one student from each team places them on a team randomly

### PR DESCRIPTION
**Pull request by justino599**
This mutation takes one student from each team and swaps them

Closes #392

### CODE REVIEW

File: api/ai/priority_algorithm/mutations/random_slice.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            picked_students.append(
+                team.student_ids[random.randrange(len(team.student_ids))]
+            )
+
+        # Shuffle picked students
+        shuffle(picked_students)
+
+        # Put them back into the teams
+        for team in available_priority_teams:
+            team.student_ids.remove(picked_students.pop())
```
Comment (Opeyem1a) [inline api/ai/priority_algorithm/mutations/random_slice.py:31]: I'm confused, maybe the comments are out of date?

Comment (Opeyem1a) [reply api/ai/priority_algorithm/mutations/random_slice.py:31]: Like why does this remove?

File: api/ai/priority_algorithm/mutations/random_slice.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    available_priority_teams = get_available_priority_teams(priority_team_set)
+    try:
+        if len(available_priority_teams) < 2:
+            return priority_team_set
+
+        # Pop a random student from each team
+        picked_students = []
+        for team in available_priority_teams:
+            picked_students.append(
+                team.student_ids[random.randrange(len(team.student_ids))]
```
Comment (Opeyem1a) [inline api/ai/priority_algorithm/mutations/random_slice.py:23]: not seeing how this pops a student from each team ngl

**Review by Opeyem1a — COMMENTED** [review]
Tests would be fire

**Review by Opeyem1a — CHANGES_REQUESTED** [review]
Oops forgot to mark as request changes

**Comment by justino599** [comment]
I actually apologize for this code, those were very dumb mistakes. They are fixed now.

**Pull request merged**